lib/hexapdf/type/image.rb

```diff
@@ -254,6 +257,48 @@ def png_chunk(type, data = '')
         [data.length].pack("N") << type << data << [Zlib.crc32(data, Zlib.crc32(type))].pack("N")
       end
 
+      # Writes the image as TIFF to the given IO stream.
+      def write_tiff(io, info)
+        stream = stream_source
+
+        # header
+        io << ['49492A00'].pack('H*')
+
+        # offset to IFD
+        pad_size = stream.length % 2
+        io << [stream.length + 8 + pad_size].pack('L')
```

Comment:
Using `stream.length` is only possible when the PDF file was actually read which *should* be the case here (because if we create a PDF with a TIFF image, why would we use this way to save the TIFF image?). I guess it's okay as we can always change this later if there really is a need.

According to the TIFF spec, using a header of `4949` means little-endian encoding. Therefore you need to use `L<`, otherwise the native encoding is used.

I would also change the order to `8 + ...` so  that it's clearer that the `8` refers to the header bytes.